src/build.cc

```diff
@@ -1014,17 +1019,38 @@ bool Builder::ExtractDeps(CommandRunner::Result* result,
     if (content.empty())
       return true;
 
-    DepfileParser deps(config_.depfile_parser_options);
-    if (!deps.Parse(&content, err))
+    if (deps_type == "gcc") {
+      DepfileParser deps(config_.depfile_parser_options);
+      if (!deps.Parse(&content, err))
+        return false;
+
+      // XXX check depfile matches expected output.
+      deps_nodes->reserve(deps.ins_.size());
+      for (vector<StringPiece>::iterator i = deps.ins_.begin();
```

Comment:
Can you use a for-range loop for simplicity and add a comment about the subtle in-place modification issue here, e.g.:

```
for (StringPiece input : deps.ins_) {
  uint64_t slash_bits;
  // NOTE: In-place modification of deps buffer.
  // This is o as nothing else will use it after this loop.
  CanonicalizePath(const_cast<char*>(input.str_), input.len_, &slash_bits);
  deps_nodes->push_back(state_->GetNode(input, slash_bits));
}
```

Reply:
This was existing code I re-indented.  I've made this change in a dedicated preceding commit with you as the author.